### PR TITLE
Page through the results instead of taking only page 1

### PR DESCRIPTION
**Pull request by smn**
Currently the report will only grab the first 1K results because that's what the client we're using will default to.

### CODE REVIEW

**Comment by smn** [comment]
@DevChima ready for review

**Comment by smn** [comment]
@rudigiesler and @jpaidoussi ready for review. I would actually like to spend some time to add this to the client library instead of only here.

File: ci/management/commands/generate_reports.py [hunk 1]
```diff
@@ -141,14 +152,19 @@ def get_identity(self, ids_client, identity):
         self.identity_cache[identity] = identity_object
         return identity_object
 
+    def get_registrations(self, hub_client, **kwargs):
+        registrations = hub_client.get_registrations(kwargs)
+        cursor = registrations['next']
+        while cursor:
+            yield registrations
```
Comment (rudigiesler) [inline ci/management/commands/generate_reports.py]: If we do a

```
for r in registrations.get('results', []):
    yield r
```

over here, that will elimate the need for the double loop down below

File: ci/tests.py [hunk 2]
```diff
@@ -82,6 +94,7 @@ def test_generate_report(self):
             'http://hub/registrations/',
             json={
                 'count': 1,
+                'next': None,
```
Comment (rudigiesler) [inline ci/tests.py:115]: This won't actually test that we're getting the next page successfully. Maybe we should have 2 results split over 2 pages, and ensure that we get both results in the export.

**Comment by smn** [comment]
@rudigiesler @jpaidoussi and @DevChima ready for review again

File: ci/management/commands/generate_reports.py [hunk 3]
```diff
@@ -141,14 +152,21 @@ def get_identity(self, ids_client, identity):
...
+        registrations = hub_client.get_registrations(kwargs)
+        cursor = registrations['next']
+        while cursor:
+            for result in registrations['results']:
+                yield result
+            params = parse_cursor_params(cursor)
+            registrations = hub_client.get_registrations(params)
+            cursor = registrations['next']
+        for result in registrations['results']:
+            yield result
```
Comment (smn) [inline ci/management/commands/generate_reports.py:165]: I'm sure there's a nicer way of doing this but I'm leaving that investigation for when I start implementing this in the `seed-services-client` code base.

Comment (rudigiesler) [reply ci/management/commands/generate_reports.py:165]: If you get the registrations at the beginning of the loop, and then yield the registrations last, then we won't need this extra step (I think I did this before with a `while True`, and then a `break` if `next` is `None`). But we can look into that when we implement this in the client.

**Review by rudigiesler — APPROVED** [review]

**Pull request merged**